### PR TITLE
Add line input

### PR DESCRIPTION
**Pull request by mattnibs**
Add support for a line input option to zq that translates each line from an input as a zed string.

Closes #4175

### CODE REVIEW

File: zio/lineio/ztests/reader.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+zed: |
+  yield regexp(/\[([0-9]+)\] = (\w+)/, this)
+  | this != null
+  | yield {port:uint16(this[1]),cipher:this[2]}
```
Comment (philrz) [inline zio/lineio/ztests/reader.yaml:4]:
```suggestion
  | yield {suite:uint16(this[1]),cipher:this[2]}
```

File: zio/lineio/ztests/reader.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  {
+  [49307] = TLS_ECDHE_PSK_WITH_CAMELLIA_256_CBC_SHA384,
+  [49171] = TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,
+  [49286] = TLS_ECDHE_ECDSA_WITH_CAMELLIA_128_GCM_SHA256,
+  }
+
+output: |
+  {port:49307(uint16),cipher:"TLS_ECDHE_PSK_WITH_CAMELLIA_256_CBC_SHA384"}
+  {port:49171(uint16),cipher:"TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA"}
+  {port:49286(uint16),cipher:"TLS_ECDHE_ECDSA_WITH_CAMELLIA_128_GCM_SHA256"}
```
Comment (philrz) [inline zio/lineio/ztests/reader.yaml:18]:
```suggestion
  {suite:49307(uint16),cipher:"TLS_ECDHE_PSK_WITH_CAMELLIA_256_CBC_SHA384"}
  {suite:49171(uint16),cipher:"TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA"}
  {suite:49286(uint16),cipher:"TLS_ECDHE_ECDSA_WITH_CAMELLIA_128_GCM_SHA256"}
```

[the remark this answers is not in the document]
Comment (philrz) [reply zio/lineio/ztests/reader.yaml:18]: Indeed, they're actually identifiers for cipher suites, and that's what Wireshark calls them too.

**Review by philrz — COMMENTED** [review]
I tested it out on the branch and it works great! I'll let the code reviewers give the final approval, but it's an enthusiastic functional 👍 from me. Thanks @mattnibs!

File: zio/lineio/ztests/reader.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+zed: |
+  yield regexp(/\[([0-9]+)\] = (\w+)/, this)
+  | this != null
+  | yield {suite:uint16(this[1]),cipher:this[2]}
```
[the remark this answers is not in the document]
Comment (mattnibs) [reply zio/lineio/ztests/reader.yaml:4]: ok

**Review by nwt — APPROVED** [review]
:shipit:

**Pull request merged**